tests/acceptance/pageObjects/FilesPageElement/filesList.js

```diff
@@ -441,24 +441,37 @@ module.exports = {
               if (attr.value.indexOf('uk-invisible') >= 0) {
                 return
               }
-              this.api.elementIdAttribute(element.ELEMENT, 'aria-label', attr => {
-                switch (attr.value) {
-                  case 'Directly shared with people':
-                    indicators.push('user-direct')
-                    break
-                  case 'Shared with people through one of the parent folders':
-                    indicators.push('user-indirect')
-                    break
-                  case 'Directly shared with links':
-                    indicators.push('link-direct')
-                    break
-                  case 'Shared with links through one of the parent folders':
-                    indicators.push('link-indirect')
-                    break
-                  default:
-                    console.warn('Unknown share indicator found: "' + attr + '"')
+              const setIndicatorValue = () => {
+                this.api.elementIdAttribute(element.ELEMENT, 'aria-label', attr => {
+                  switch (attr.value) {
+                    case 'Directly shared with people':
+                      indicators.push('user-direct')
+                      break
+                    case 'Shared with people through one of the parent folders':
+                      indicators.push('user-indirect')
+                      break
+                    case 'Directly shared with links':
+                      indicators.push('link-direct')
+                      break
+                    case 'Shared with links through one of the parent folders':
+                      indicators.push('link-indirect')
+                      break
+                    default:
+                      console.warn('Unknown share indicator found: "' + attr + '"')
+                  }
+                })
+              }
+              // Sometimes it takes a bit long for the sharing indicator element to be displayed properly.
+              // If the element is not displayed, we wait for 2 seconds and then check it again.
+              // This process is repeated 3 times.
+              if (attr.value) {
+                setIndicatorValue()
+              } else {
+                for (let i = 0; i < 3; i++) {
+                  this.pause(2000)
```

Comment:
You're adding 6 seconds pause if they are not visible on the first attempt. There is no condition to stop waiting if the indicators were found in the first or second retry.

Reply:
yes, generally retry is maybe a good thing, but let's have a timeout and a shorted retry period
usually we use `globals.waitForConditionTimeout` as timeout and `globals.waitForConditionPollInterval` as the time between retries
both defined in https://github.com/owncloud/web/blob/master/nightwatch.conf.js